doc/design/logs.md

```diff
@@ -0,0 +1,35 @@
+# Logs in SQLFlow
+## Motivation
+In order to know well about the runtime status of the SQLFlow job, we need to count the number of `TRAIN/PREDICT/EXPLAIN/NORMAL` tasks over a period of time. For example, using ELK Stack for log query and analysis‎. Generally, such statistics components are implemented by logs.
+
+## logging library
+Several statistics components use `logtail` to collect logs. The `logtail` demands a file. If such information is written to `stdout`, we should redirect the `stdout` to a local file, the file size will increase until the SQLFlow server terminates, so we need a rolling file to store the log. Besides, a well-formatted log is readable.
+We use [Logrus](https://github.com/sirupsen/logrus) for layout, combining with [https://github.com/natefinch/lumberjack] for rolling the log file.
+
```

Comment:
There are several places contains the logging of the standard package, are we changing them to `logrus`?

https://github.com/sql-machine-learning/sqlflow/blob/e7142c09b6449905fb4ea11bff764f4de982d69a/pkg/sql/executor_ir.go#L65

Reply:
Sure, let's make the change.
In fact, I'm not sure the one who submits the SQL program by `log.Printf("runSQLProgram error: %v", err)`. It's a little hard to debug in lots of logs.

Reply:
Right. Let's remove this `log.Printf`.